presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushTableWriteThroughUnion.java

```diff
@@ -39,6 +39,12 @@
     private static final Capture<UnionNode> CHILD = newCapture();
 
     private static final Pattern<TableWriterNode> PATTERN = tableWriterNode()
+            // The primary incentive of this optimizer is to increase the parallelism for table
+            // write. For a table with partitioning scheme, parallelism for table writing is
+            // guaranteed regardless of this optimizer. The level of local parallelism will be
+            // determined by LocalExecutionPlanner separately, and shouldn't be a concern of
+            // this optimizer.
```

Comment:
When I read the comment, I had the impression it does a subtle hint at what's written under it. It's been an implementation comment, I'd prefer to leave it this way.